src/codegen/llvm/codegen_llvm_visitor.cpp

```diff
@@ -54,11 +54,15 @@ static bool can_vectorize(const ast::CodegenForStatement& statement, symtab::Sym
             return false;
     }
 
-    // Check there is no control flow in the kernel.
-    const std::vector<ast::AstNodeType> unsupported_nodes = {ast::AstNodeType::IF_STATEMENT};
-    const auto& collected = collect_nodes(statement, unsupported_nodes);
+    // Check for simple supported control flow in the kernel (single if/else statement).
+    const std::vector<ast::AstNodeType> supported_control_flow = {ast::AstNodeType::IF_STATEMENT};
+    const auto& supported = collect_nodes(statement, supported_control_flow);
```

Comment:
what happens if there is `if` within `if`? (I should check myself! )

Reply:
IR would be wrong I think😅 I have removed the use of the vectorisation check in the for loop, so this function is not really used anymore.

I think a more robust version is needed where we check for nested ifs